types.ts

```diff
@@ -187,95 +234,202 @@ interface BaseAuthentication {
+   * This is typically just a base url with no parameters. Coda will append the `scope`
+   * parameter automatically. If the authorization flow requires additional parameters,
+   * they may be specified using {@link additionalParams}.
+   */
   authorizationUrl: string;
+  /**
+   * The URL that Coda will hit in order to exchange the temporary token for an access token
+   * at the end of the OAuth handshaek flow.
+   */
   tokenUrl: string;
+  /**
+   * Scopes that are required to use this pack.
+   */
   scopes?: string[];
+  /**
+   * A custom prefix to be used when passing the access token in the HTTP Authorization
+   * header when making requests. Typically this prefix is `Bearer` which is what will be
+   * used if this value is omitted. However, some services require a different prefix.
+   * When sending authenticated requests, a HTTP header of the form
+   * `Authorization: <tokenPrefix> <token>` will be used.
+   */
   tokenPrefix?: string;
+  /**
+   * Option custom URL parameters and values that should be included when redirecting the
+   * user to the {@link authorizationUrl}.
+   */
   additionalParams?: {[key: string]: any};
 
-  // Some OAuth providers will return the API domain with the OAuth response.
-  // This is the key in the OAuth response json body that points to the endpoint.
+  /**
+   * In rare cases, OAuth providers will return the specific API endpoint domain for the user as
+   * part of the OAuth token exchange response. If so, this is the property in the OAuth
+   * token exchange response JSON body that points to the endpoint.
+   */
   endpointKey?: string;
 
-  // Some OAuth providers require passing the token as a URL param.
+  /**
+   * In rare cases, OAuth providers ask that a token is passed as a URL parameter
+   * rather than an HTTP header. If so, this is the name of the URL query parameter
+   * that should contain the token.
+   */
   tokenQueryParam?: string;
 }
 
+/**
+ * Authenticate using HTTP Basic authorization. The user provides a username and password
+ * (sometimes optional) which are included as an HTTP header according to the Basic auth standard.
+ *
+ * See https://en.wikipedia.org/wiki/Basic_access_authentication
+ */
 export interface WebBasicAuthentication extends BaseAuthentication {
   type: AuthenticationType.WebBasic;
+  /**
+   * Configuration for labels to show in the UI when the user sets up a new acount.
+   */
   uxConfig?: {
+    /**
+     * A placeholder value for the text input where the user will enter a username.
+     */
     placeholderUsername?: string;
+    /**
+     * A placeholder value for the text input where the user will enter a password.
+     */
     placeholderPassword?: string;
 
-    // Some auth providers pass apiKeys in the username and do not require a password
+    /**
+     * If true, only a username input will be shown to the user.
+     * Some services pass API keys in the username field and do not require a password
+     * (not recommended!).
```

Comment:
I feel like this could confuse the developer reading this, since they are likely stuck with whatever the API provider has chosen and can't use an alternative. I'd suggest removing.